src/components/TextField/TextField.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { Component } from 'react'
+import Form from 'react-bootstrap/Form'
+import { Props } from './interfaces'
+
+/**
+ * A customizable text field component. It's a wrapper component built upon react's form controls.
+ */
+class TextField extends Component<Props, { textFieldValue: string }> {
+  constructor(props: Props) {
+    super(props)
+
+    this.state = {
+      textFieldValue: this.props.value ? this.props.value : '',
+    }
+  }
+
+  handleInputChange = (event: any) => {
+    this.setState({
+      textFieldValue: event.currentTarget.value,
+    })
+  }
```

Comment:
Instead of `TextField` defining its own `onChange` handler, should we expose this via a prop so the component that is using `TextField` can define what should happen `onChange`?